pkg/apis/migration/v1alpha1/migcluster_types.go

```diff
@@ -498,3 +502,89 @@ func (r *MigCluster) UpdateProvider(provider pvdr.Provider) {
 		}
 	}
 }
+
+func (m *MigCluster) GetInternalRegistryPath(c k8sclient.Client) (string, error) {
+	client, err := m.GetClient(c)
+	if err != nil {
+		return "", err
+	}
+	isList := imgapi.ImageStreamList{}
+	err = client.List(
+		context.TODO(),
+		k8sclient.InNamespace("openshift"),
+		&isList)
+	if err == nil && len(isList.Items) > 0 {
+		if value := isList.Items[0].Status.DockerImageRepository; len(value) > 0 {
+			ref, err := reference.Parse(value)
+			if err == nil {
+				return ref.Registry, nil
+			}
+		}
+	}
+	if client.MajorVersion() != 1 {
+		return "", errors.New(fmt.Sprintf("server version %v.%v not supported. Must be 1.x", client.MajorVersion(), client.MinorVersion()))
+	}
+	if client.MinorVersion() < 7 {
+		return "", errors.New(fmt.Sprintf("Kubernetes version 1.%v not supported. Must be 1.7 or greater", client.MinorVersion()))
+	} else if client.MinorVersion() <= 11 {
+		registrySvc := kapi.Service{}
+		err := client.Get(
+			context.TODO(),
+			k8sclient.ObjectKey{
+				Namespace: "default",
+				Name:      "docker-registry",
+			},
+			&registrySvc)
+		if err != nil {
+			// Return empty registry host but no error; registry not found
+			return "", nil
+		}
+		internalRegistry := registrySvc.Spec.ClusterIP + ":" + strconv.Itoa(int(registrySvc.Spec.Ports[0].Port))
+		return internalRegistry, nil
+	} else {
+		config := kapi.ConfigMap{}
```

Comment:
I think a more authoritative place to get this would be from the apiserver observed config

```
$ oc get openshiftapiservers.operator.openshift.io -oyaml | grep image
            f:imagePolicyConfig: {}
      imagePolicyConfig:
        internalRegistryHostname: image-registry.openshift-image-registry.svc:5000
```

Reply:
Currently this is doing it in the same way the plugin handles it. If we do end up needing to make changes, we should probably make similar changes there.
@dymurray We went through several options trying to find the most reliable way of doing this when we added this to the plugin last year. Do you remember whether we considered this other option as well?

Reply:
I think it makes sense to grab this from the CR. Seems to be the source of truth.

Reply:
Question: does this change force us to run the controller on OCP 4?  I doubt `openshiftapiservers.operator.openshift.io` exists on OCP 3

Reply:
@djwhatle This is code we'll only run if it's OCP4 (there's a check that MinorVersion is 12 or above) -- the ocp3 fallback is just using the cluster IP, since the equivalent in OCP3 isn't available via API at all. Note that both of these are fallbacks if the primary means of lookup fails -- if there's at least one imagestream in the "openshift" namespace, we parse the registry location from that metadata.